pkgs/development/tools/regclient/default.nix

```diff
@@ -20,22 +27,56 @@ buildGoModule rec {
   ldflags = [
     "-s"
     "-w"
-    "-X main.VCSTag=${tag}"
+    "-X github.com/regclient/regclient/internal/version.vcsTag=${tag}"
   ];
 
-  postInstall =
-    lib.concatStringsSep "\n" (
-      map (bin: ''
-        mkdir -p ''$${bin}/bin &&
-        mv $out/bin/${bin} ''$${bin}/bin/ &&
-        ln -s ''$${bin}/bin/${bin} $out/bin/
-      '') bins
-    );
+  nativeBuildInputs = [ installShellFiles ];
+
+  postInstall = lib.concatStringsSep "\n" (
+    map
+      (bin: ''
+        export bin=''$${bin}
+        export outputBin=bin
+
+        mkdir -p $bin/bin
+        mv $out/bin/${bin} $bin/bin
+        ln -s $bin/bin/${bin} $out/bin
+
+        $bin/bin/${bin} completion bash >${bin}.bash
+        $bin/bin/${bin} completion fish >${bin}.fish
+        $bin/bin/${bin} completion zsh >${bin}.zsh
+        installShellCompletion ${bin}.{bash,fish,zsh}
+
+        mkdir -p $out/share/zsh/site-functions
+        mkdir -p $out/share/fish/vendor_completions.d
+        mkdir -p $out/share/bash-completion/completions
+        ln -s $bin/share/zsh/site-functions/_${bin} $out/share/zsh/site-functions
+        ln -s $bin/share/fish/vendor_completions.d/${bin}.fish $out/share/fish/vendor_completions.d
+        ln -s $bin/share/bash-completion/completions/${bin}.bash $out/share/bash-completion/completions
+
+        unset bin outputBin
```

Comment:
We should probably use something like lndir to create the combined package instead of manually doing it

Reply:
lndir is a command provided by xorg.lndir which copies a directory into a symlink tree. see maybe as an example https://github.com/NixOS/nixpkgs/blob/2c6e758797682f0915ec290b86a7787f0faa8918/pkgs/applications/editors/neovim/wrapper.nix#L166